packages/edit-site/src/components/header/index.js

```diff
@@ -103,6 +109,31 @@ export default function Header( {
 
 	const isFocusMode = templateType === 'wp_template_part';
 
+	let displayTitle;
+
+	if ( entityTitle === 'Index' || entityTitle === 'Home' ) {
```

Comment:
Gut says this is not correct and depends on what should be displayed when the editor is opened via `Appearance >templates` / `Appearance >template parts`

Reply:
> Gut says this is not correct and depends on what should be displayed when the editor is opened via Appearance >templates / Appearance >template parts

When accessing a template via Appearance > Templates, it seems fair to continue displaying the template name.

Reply:
Except that path does not open in the Site Editor but the block editor. That still needs to be solved.

Reply:
Indeed 🙈